fix: propagate providerData for function_calls in chat completions converter

danielmklein opened the pull request:
We do not currently propagate the provider data on `function_call` items in the chat completions converter -- this can be problematic in cases when a user wants to pass through provider-specific data (especially given that we construct an `assistant` message in one doesn't exist).

seratch commented:
The test code compilation fails. Can you fix it?

```
Error: packages/agents-openai build-check: test/openaiChatCompletionsConverter.test.ts(384,7): error TS2352: Conversion of type '{ type: "function_call_result"; id: string; callId: string; output: string; providerData: { extra: string; }; }' to type '{ type: "function_call_result"; status: "in_progress" | "completed" | "incomplete"; name: string; output: string | { type: "text"; text: string; providerData?: Record<string, any> | undefined; } | { ...; } | { ...; } | ({ ...; } | ... 1 more ... | { ...; })[]; callId: string; providerData?: Record<...> | undefined; ...' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
```

danielmklein commented:
> The test code compilation fails. Can you fix it?
> 
> ```
> Error: packages/agents-openai build-check: test/openaiChatCompletionsConverter.test.ts(384,7): error TS2352: Conversion of type '{ type: "function_call_result"; id: string; callId: string; output: string; providerData: { extra: string; }; }' to type '{ type: "function_call_result"; status: "in_progress" | "completed" | "incomplete"; name: string; output: string | { type: "text"; text: string; providerData?: Record<string, any> | undefined; } | { ...; } | { ...; } | ({ ...; } | ... 1 more ... | { ...; })[]; callId: string; providerData?: Record<...> | undefined; ...' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
> ```

@seratch sorry for wasting a bit of your time there! I've fixed that type issue and updated `CONTRIBUTING.md` to add the `build-check` call to the command in the pull request steps as well.
